src/intrinsics/prepack/global.js

```diff
@@ -263,14 +263,21 @@ export default function(realm: Realm): void {
         "global.__evaluatePureFunction",
         "__evaluatePureFunction",
         0,
-        (context, [functionValue]) => {
+        (context, [functionValue, callback]) => {
           invariant(functionValue instanceof ECMAScriptSourceFunctionValue);
           invariant(typeof functionValue.$Call === "function");
           let functionCall: Function = functionValue.$Call;
           return realm.evaluatePure(
             () => functionCall(realm.intrinsics.undefined, []),
             /*bubbles*/ true,
-            /*reportSideEffectFunc*/ null
+            /*reportSideEffectFunc*/ callback === undefined || callback === realm.intrinsics.undefined
+              ? null
+              : () => {
+                  invariant(callback instanceof ECMAScriptSourceFunctionValue);
```

Comment:
Why is this an invariant? I.e. do you believe that Prepack's code is incorrect if callback is something else? If so, why?

Reply:
I believe I fixed this in a subsequent PR.